### PR TITLE
add `compiled_contracts` artifacts constraints

### PR DESCRIPTION
**Pull request by marcocastignoli**
See https://github.com/verifier-alliance/database-specs/pull/12

### CODE REVIEW

File: services/database/migrations/20240729112258-compiled_contracts-constraints.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+"use strict";
```
Comment (kuzdogan) [inline services/database/migrations/20240729112258-compiled_contracts-constraints.js:1]:
```suggestion
// Constraints for the `compiled_contracts` table, according to the Verifier Alliance schema https://github.com/verifier-alliance/database-specs/pull/12
"use strict";
```

File: services/database/migrations/20240729112258-compiled_contracts-constraints.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+exports.down = function (db, callback) {
+  async.series(
+    [
+      db.runSql.bind(
+        db,
+        `
+        ALTER TABLE compiled_contracts DROP CONSTRAINT compiled_contracts_pseudo_pkey;
+        
+        ALTER TABLE compiled_contracts
+        ADD CONSTRAINT compiled_contracts_pseudo_pkey UNIQUE (compiler, language, creation_code_hash, runtime_code_hash);
```
Comment (kuzdogan) [inline services/database/migrations/20240729112258-compiled_contracts-constraints.js:93]: Why do we add and remove the constraint here?

**Review by kuzdogan — COMMENTED** [review]
I haven't looked in detail but I guess it's a copy paste of the VerA code?

In that case let's wait for that PR before merging this

File: services/database/migrations/20240729112258-compiled_contracts-constraints.js (hunk 2 above)
Comment (marcocastignoli) [reply services/database/migrations/20240729112258-compiled_contracts-constraints.js:93]: down migration is used to restore the state to the previous one. So I'm removing the new constraint and adding back the old one.

**Review by kuzdogan — APPROVED** [review]

**Pull request merged**